### DISCUSSION TITLE
download project to device

### DISCUSSION THREAD

**Discussion by Amy15Fee** [gijzelaerr/python-snap7#422] [Q&A] [answered]
Hello,

i just wanted to ask if it is possible to download a project to the device with this python wrapper? 
I'm trying to write some tests and this feature would help a lot to automate the tests later on.

**Comment by mcres** [comment, marked as answer]
I think you are looking for the [TIA Portal Openness API](https://support.industry.siemens.com/cs/document/108716692/tia-portal-openness-introduction-and-demo-application?dti=0&lc=en-US)

**Comment by gijzelaerr** [comment]
Closing as answered. Downloading TIA Portal projects to a PLC is outside the scope of snap7. The TIA Portal Openness API was suggested as an alternative.
